src/web/app/pages-instructor/instructor-student-list-page/instructor-student-list-page.component.ts

```diff
@@ -67,461 +35,116 @@ interface GetCourseDetailsResponse {
-   * Get the list of students which need to be hidden on the StudentTable display.
-   */
-  getStudentsToHide(courseId: string): string[] {
-    return this.studentDetailsList
-        .filter((student: StudentDetails) => student.courseId === courseId && !student.isChecked)
-        .map((student: StudentDetails) => student.email);
-  }
-
-  /*------------------------------Functions to formulate data for student list from item------------------------*/
-  /**
-   * Formulate data for student list.
-   */
-  getStudentListSectionDataForCourse(courseId: string): StudentListSectionData[] {
-    return this.courseStudentListSectionDataMap[courseId] || [];
-  }
-
-  /**
-   * Formulate data for student list.
-   */
-  getStudentListSectionDataFromCourseDetails(courseDetails: CourseDetails): StudentListSectionData[] {
-    const sections: StudentListSectionData[] = [];
-    courseDetails.sections.forEach((section: SectionDetails) => {
-      const sectionData: StudentListSectionData = this.mapSectionForStudentList(section);
-      sections.push(sectionData);
+   * Loads students of a specified course.
+   */
+  loadStudents(courseTab: CourseTab): void {
+    this.studentService.getStudentsFromCourse(courseTab.course.courseId)
+        .subscribe((students: Students) => {
+          const sections: StudentIndexedData = students.students.reduce((acc: StudentIndexedData, x: Student) => {
+            const term: string = x.sectionName;
+            (acc[term] = acc[term] || []).push(x);
+            return acc;
+          }, {});
+
+          const teams: StudentIndexedData = students.students.reduce((acc: StudentIndexedData, x: Student) => {
+            const term: string = x.teamName;
+            (acc[term] = acc[term] || []).push(x);
+            return acc;
+          }, {});
+
+          courseTab.stats = {
+            numOfStudents: students.students.length,
+            numOfSections: Object.keys(sections).length,
+            numOfTeams: Object.keys(teams).length,
+          };
+
+          Object.keys(sections).forEach((key: string) => {
+            const studentsInSection: Student[] = sections[key];
+
+            const data: StudentListStudentData[] = [];
+            studentsInSection.forEach((student: Student) => {
+              const studentData: StudentListStudentData = {
+                name : student.name,
+                status : (student.joinState === JoinState.JOINED) ? 'Joined' : 'Yet to Join',
+                email : student.email,
+                team : student.teamName,
+              };
+              data.push(studentData);
+            });
+
+            this.loadPrivilege(courseTab, key, data);
+          });
+        }, (resp: ErrorMessageOutput) => { this.statusMessageService.showErrorMessage(resp.error.message); });
+  }
+
+  /**
+   * Loads privilege of an instructor for a specified course and section.
+   */
+  loadPrivilege(courseTab: CourseTab, sectionName: string, students: StudentListStudentData[]): void {
```

Comment:
This is a bit expensive? Consider using a cache?

Reply:
A cache is useful if Im repeatedly calling this function for the same states. However, it does not seem to be the case here right?

I break down a course into its sections and call this API for each section.